spec/GraphQLOverHTTP.md

```diff
@@ -187,94 +305,298 @@ This request could be sent via an HTTP GET as follows:
 http://example.com/graphql?query=query(%24id%3A%20ID!)%7Buser(id%3A%24id)%7Bname%7D%7D&variables=%7B%22id%22%3A%22QVBJcy5ndXJ1%22%7D
 ```
 
-Note: {query} and {operationName} parameters are encoded as raw strings in the query component. Therefore if the query string contained `operationName=null` then it should be interpreted as the {operationName} being the string `"null"`. If a literal `null` is desired, the parameter (e.g. {operationName}) should be omitted.
-
-GET requests MUST NOT be used for executing mutation operations. If the values of {query} and {operationName} indicate that a mutation operation is to be executed, the server SHOULD immediately respond with error status code `405` (Method Not Allowed) and halt execution. This restriction is necessary to conform with the long-established semantics of safe methods within HTTP.
+Note: {query} and {operationName} parameters are encoded as raw strings in the
+query component. Therefore if the query string contained `operationName=null`
+then it should be interpreted as the {operationName} being the string `"null"`.
+If a literal `null` is desired, the parameter (e.g. {operationName}) should be
+omitted.
 
+GET requests MUST NOT be used for executing mutation operations. If the values
+of {query} and {operationName} indicate that a mutation operation is to be
+executed, the server MUST respond with error status code `405` (Method Not
+Allowed) and halt execution. This restriction is necessary to conform with the
+long-established semantics of safe methods within HTTP.
 
 ## POST
 
-A GraphQL POST request instructs the server to perform a query or mutation operation. A GraphQL POST request MUST have a body which contains values of the request parameters encoded according to the value of `Content-Type` header of the request.
+A GraphQL POST request instructs the server to perform a query or mutation
+operation. A GraphQL POST request MUST have a body which contains values of the
+request parameters encoded in one of the officially recognized GraphQL media
+types, or another media type supported by the server.
```

Comment:
I see your point. I think that though JSON is the concrete representation, because there are two distinct media types (`application/json` and `application/graphql+json`) it makes sense to keep it as is - but if you can think of better wording I'm definitely open to it!